Cloud backend issues

MrFlashAccount opened the pull request:
### Pull Request Description

Closes: enso-org/cloud-v2#1481

### Important Notes

<!--
- Mention important elements of the design.
- Mention any notable changes to APIs.
-->

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [ ] Unit tests have been written where possible.

MrFlashAccount committed:
🐞Box should be full width

MrFlashAccount committed:
🐞☁️Can’t edit a secret.

MrFlashAccount committed:
🐞☁️Clicking out the datalink config made it close without warning or saving.

MrFlashAccount committed:
☁️Right click on labels shows in wrong place - should be below the chosen label, left aligned box with label.

MrFlashAccount committed:
🐞☁️On team space I can't Move to trash when selected multiple assets

MrFlashAccount committed:
🐞☁️Edit description from right click menu (on secret) didnt cause the properties to be invalidated, needed to choose file and move back.

MrFlashAccount committed:
Fix tests

MrFlashAccount committed:
Fix types

MrFlashAccount committed:
Merge branch 'develop' into wip/sergeigarin/fix-remote-backend-issues

# Conflicts:
#	app/dashboard/src/components/AriaComponents/Inputs/DatePicker/DatePicker.tsx
#	app/dashboard/src/modals/UpsertSecretModal.tsx

somebody1234 commented on app/dashboard/src/components/JSONSchemaInput.tsx:
```diff
@@ -92,7 +92,7 @@ export default function JSONSchemaInput(props: JSONSchemaInputProps) {
                   size={1}
                   className={twMerge(
-                    'focus-child text w-60 grow rounded-input border-0.5 bg-transparent px-input-x read-only:read-only',
+                    'focus-child text w-60 w-full grow rounded-input border-0.5 bg-transparent px-input-x read-only:read-only',
```
redundant `w-60`

somebody1234 commented on app/dashboard/src/components/JSONSchemaInput.tsx:
```diff
@@ -139,7 +139,7 @@ export default function JSONSchemaInput(props: JSONSchemaInputProps) {
                 size={1}
                 className={twMerge(
-                  'focus-child min-6- text40 w-60 grow rounded-input border-0.5 bg-transparent px-input-x read-only:read-only',
+                  'focus-child min-6- text40 w-full grow rounded-input border-0.5 bg-transparent px-input-x read-only:read-only',
```
oops, i think `min-6- text40` is broken... we probably want to check in blame for the original correct code

MrFlashAccount replied:
![CleanShot 2024-09-16 at 10 11 55@2x](https://github.com/user-attachments/assets/fefdcf18-f824-46d2-bce6-eb786896f22c)

I think it was a bad regex: https://github.com/enso-org/enso/pull/10715/files#diff-714d58b65c74fc059da856df96d3e22e551c9aa5de3396fd7de6a52a50b120b3R157

somebody1234 replied:
hmm. we may want to enable the invalid tailwind classes lint. thoughts?

MrFlashAccount replied:
agree

somebody1234 commented on app/dashboard/src/components/JSONSchemaInput.tsx:
```diff
@@ -92,7 +92,7 @@ export default function JSONSchemaInput(props: JSONSchemaInputProps) {
                   size={1}
                   className={twMerge(
-                    'focus-child text w-60 grow rounded-input border-0.5 bg-transparent px-input-x read-only:read-only',
+                    'focus-child text w-60 w-full grow rounded-input border-0.5 bg-transparent px-input-x read-only:read-only',
```
also with the changes in this file i guess make sure nested objects (nested divs) still look fine instead of e.g. overflowing

MrFlashAccount replied:
I tested the datalink dialog and everything seems to be correct. Do we use this Input somewhere else?

somebody1234 replied:
nope! if everything looks fine there then it's all good

somebody1234 commented on app/dashboard/src/components/dashboard/AssetRow.tsx:
```diff
@@ -250,25 +253,25 @@ export const AssetRow = React.memo(function AssetRow(props: AssetRowProps) {
-                setAsset(object.merger({ description: asset.description }))
-                throw error
-              })
+            // eslint-disable-next-line no-restricted-syntax
```
what is the restricted syntax being used here?

MrFlashAccount replied:
Early returns. Maybe we should allow them?

somebody1234 replied:
ah. possibly, yeah

somebody1234 commented on app/dashboard/src/layouts/AssetsTableContextMenu.tsx:
```diff
@@ -72,10 +72,18 @@ export default function AssetsTableContextMenu(props: AssetsTableContextMenuProp
     !isCloud ||
     Array.from(selectedKeys, (key) => {
       const userPermissions = nodeMapRef.current.get(key)?.item.permissions
+      const selfGroupPermission = userPermissions?.find(
```
prefer `permissions.tryFindSelfPermission` instead of this entire function

somebody1234 commented on app/dashboard/src/providers/ModalProvider.tsx:
```diff
@@ -37,23 +39,34 @@ export interface ModalProviderProps extends Readonly<React.PropsWithChildren> {}
 export default function ModalProvider(props: ModalProviderProps) {
   const { children } = props
   const [modal, setModal] = React.useState<Modal | null>(null)
+  // we use key to tell react to invaldidate the modal when we change it.
```
`we` -> `We`

MrFlashAccount committed:
Fix lint

somebody1234 commented on app/dashboard/src/providers/ModalProvider.tsx:
```diff
@@ -37,23 +39,34 @@ export interface ModalProviderProps extends Readonly<React.PropsWithChildren> {}
   const { children } = props
   const [modal, setModal] = React.useState<Modal | null>(null)
+  // we use key to tell react to invaldidate the modal when we change it.
+  const [key, setKey] = React.useState(0)
```
i prefer `useRefresh`, it's true that a numeric `key` does work for any program that runs for less than like a trillion years, but i guess it's just conceptually... less confusing? to not use a number - although it's not like we can't change `useRefresh` to use a number

MrFlashAccount replied:
`useRefresh` isn't the same. By setting a key, we tell react that these 2 nodes of `DialogTrigger` are not the same, thus forcing react to remount the component and reinitialize it, and it seems to be correct - since `DialogTrigger` and `Dialog` are 1:1 entities

somebody1234 replied:
ah right, of course. we have `uniqueString` for that but i guess that's overkill. perhaps we should replace all `uniqueString` to something simpler like this

somebody1234 reviewed (approved):
CR ✅

somebody1234 commented:
@MrFlashAccount idk if this is ready to merge, i think there are still unaddressed CR comments above

MrFlashAccount commented:
@somebody1234 sorry, missed your comment on Friday. Here's the follow-up PR that addresses issues you've raised: https://github.com/enso-org/enso/pull/11080
